test/hotspot/jtreg/compiler/integerArithmetic/DivisionByConstant.java

```diff
@@ -0,0 +1,226 @@
+
+            var test = new TestFramework(DivisionByConstant.class);
+            test.setDefaultWarmup(1);
+            test.addFlags("-DiDiv=" + iDiv, "-Di1=" + i1, "-Di2=" + i2,
+                    "-DlDiv=" + lDiv, "-Dl1=" + l1, "-Dl2=" + l2);
+            test.start();
+        }
+    }
+
+    static long logRandom(Random r, int bits) {
+        int highestBit = r.nextInt(bits);
+        long res = r.nextLong() & (-1L >>> (Long.SIZE - 1 - highestBit));
+        return res == 0 ? 1 : res;
+    }
+
+    @Run(test = {"sDivInt", "uDivInt", "sDivLong", "uDivLong"})
+    public void run() {
+        Random r = Utils.getRandomInstance();
+        for (int i = 0; i < INVOCATIONS; i++) {
+            {
+                int x;
+                if (I_HI != Integer.MAX_VALUE) {
+                    x = r.nextInt(I_LO, I_HI + 1);
+                } else if (I_LO != Integer.MIN_VALUE) {
+                    x = r.nextInt(I_LO - 1, I_HI) + 1;
+                } else {
+                    x = r.nextInt();
+                }
+                Asserts.assertEQ(sDiv(x, I_DIV), sDivInt(x));
+            }
+            {
+                int x;
+                if (I_UHI >= 0) {
+                    x = r.nextInt(-1, I_UHI) + 1;
+                } else {
+                    x = r.nextInt(Integer.MIN_VALUE, I_UHI + 1);
+                }
+                Asserts.assertEQ(uDiv(x, I_DIV), uDivInt(x));
+            }
+            {
+                long y; int x;
+                if (L_LO >= Integer.MIN_VALUE && L_HI <= Integer.MAX_VALUE) {
+                    y = r.nextLong(L_LO, L_HI + 1);
+                    x = (int)y;
+                } else {
+                    x = r.nextInt();
+                    if (x > 0) {
+                        y = L_LO - 1 + x;
+                    } else {
+                        y = L_HI + x;
+                    }
+                }
+                Asserts.assertEQ(sDiv(y, L_DIV), sDivLong(x));
+            }
+            {
+                long y; int x;
+                if (L_UHI >= 0 && L_UHI <= Integer.MAX_VALUE) {
+                    y = r.nextLong(L_UHI + 1);
+                    x = (int)y;
+                } else {
+                    x = r.nextInt();
+                    if (x > 0) {
+                        y = x;
+                    } else {
+                        y = L_UHI + x;
+                    }
+                }
+                Asserts.assertEQ(uDiv(y, L_DIV), uDivLong(x));
+            }
+        }
+    }
+
+    @DontCompile
+    static int sDiv(int x, int y) {
+        return x / y;
+    }
+
+    @DontCompile
+    static int uDiv(int x, int y) {
+        return Integer.divideUnsigned(x, y);
+    }
+
+    @DontCompile
+    static long sDiv(long x, long y) {
+        return x / y;
+    }
+
+    @DontCompile
+    static long uDiv(long x, long y) {
+        return Long.divideUnsigned(x, y);
+    }
+
+    @Test
+    @IR(failOn = IRNode.DIV)
```

Comment:
Can you maybe also add some methods with IR rules that actually have present `IRNode.DIV` etc, just as a control test?

Reply:
Done

Reply:
Where did you put them? Not in this file?

Reply:
Aha, I see now. Why not just put a positive rule with `count`?

Reply:
I think it would be more comprehensive to test for success in the absence of the property and failure when it is present. We can test with a positive rule with `count` and flip the sense of the flag but that is indifferent. For detailed transformation we have tests in `compiler.c2.irTests` already. Thanks a lot.

Reply:
Ok, leave it as is.